src/components/organisms/Accordion/AccordionHeader/AccordionHeader.tsx

```diff
@@ -44,8 +45,15 @@ const AccordionHeader: FC<IAccordionHeader> = ({ children, id, index, textSize =
   const { getHeaderProps, isActiveSection } = useAccordionContext();
   const { ref, onClick: contextOnClick, ...headerPropsRest } = getHeaderProps(id, index);
 
+  const willBecomeActive = !isActiveSection(index);
+
   const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
-    onClick && onClick(e);
+    const curried = onClick && onClick(e);
```

Comment:
Not sure if we need to curry it. I don't see any use case where you would need an event object.

Reply:
I couldn't simply pass the boolean value to the onClick function and keep TypeScript happy. I wasn't able to find my way around the types.

Reply:
this should work:
```
export interface IAccordionHeader extends Omit<HTMLAttributes<HTMLButtonElement>, 'onClick'> {
  id: string;
  index: number;
  textSize?: 'base' | 'small';
  onClick?: (willBecomeActive: boolean) => void;
}
```

Reply:
Yup typing it like this should be enough as long as you don't pass the event to the handler...
```ts
const AccordionHeader = ({ onClick = () => undefined, ...rest }) => {
// ...
const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
  // ...
  willBecomeActive && onClick()
}
```

Reply:
Yeah I omitted the type and that did the trick